sample-app/src/main/java/com/sinch/sample/numbers/NumbersSampleFlow.java

```diff
@@ -0,0 +1,392 @@
+                    .setPattern(codePattern)
+                    .build())
+            .build();
+
+    // 2. Request for available numbers using the built-in SDK method
+    // This API is following the SDK pattern for list and the server response is wrapped inside a
+    AvailableNumberListResponse availableNumbersListResponse =
+        sinchClient.numbers().available().list(availableNumbersRequestData);
+
+    // 3. Looking for first available number
+    // auto iteration other items is available for all List responses.
+    // In case of underlying API is requiring new calls (paginated results): it will be handled
+    // automatically
+    AvailableNumber availableNumber =
+        availableNumbersListResponse.stream()
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new ApiException(
+                        "No numbers are available for the number type "
+                            + availableNumbersRequestData.getType()
+                            + " in the region "
+                            + availableNumbersRequestData.getRegionCode()
+                            + " with the pattern "
+                            + availableNumbersRequestData.getNumberPattern().get().getPattern()));
+
+    // 4. return the phone number
+    String phoneNumber = availableNumber.getPhoneNumber();
+    echo("The phone number '" + phoneNumber + "' is available. Let's rent it!\n");
+
+    return phoneNumber;
+  }
+
+  void rentPhoneNumber(int step, SinchClient sinchClient, String phoneNumber) {
+
+    echoStep(step, "Rent phone number: '" + phoneNumber + "'");
+
+    // 1. Build the request data
+    var rentRequestData = AvailableNumberRentRequestParameters.builder().build();
+
+    // 2. Request to rent the number
+    ActiveNumber rentedNumber =
+        sinchClient.numbers().available().rent(phoneNumber, rentRequestData);
+
+    echo(
+        "The number "
+            + rentedNumber.getPhoneNumber()
+            + " has been rented with the 'rent' operation.The next charge date is "
+            + rentedNumber.getNextChargeDate());
+    echo("Let's use now the 'ActiveNumbers' API to see how to manage this number.\n");
+  }
+
+  String rentPhoneNumberByAny(
+      int step, SinchClient sinchClient, String regionCode, String pattern) {
+
+    echoStep(step, "Rent a phone number for region '" + regionCode + "'");
+
+    // 1. Build the request data
+    var rentAnyNumberRequestData =
+        AvailableNumberRentAnyRequestParameters.builder()
+            .setRegionCode(regionCode)
+            .setType(NumberType.LOCAL)
+            .setNumberPattern(
+                NumberPattern.builder()
+                    .setSearchPattern(SearchPattern.START)
+                    .setPattern(pattern)
+                    .build())
+            .build();
+
+    // 2. rent a number
+    ActiveNumber rentedNumber = sinchClient.numbers().available().rentAny(rentAnyNumberRequestData);
+
+    echo(
```

Comment:
You may also add the other log `echo("Let's use now the 'ActiveNumbers' API to see how to manage this number.\n");` to have the same output with `rent` and `rentAny`

Reply:
Right. Moved this log at upper level